src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -122,7 +149,7 @@ public int Execute(string[] binaryLoggerArgs, ILogger consoleLogger, bool noRest
             // Do a restore first (equivalent to MSBuild's "implicit restore", i.e., `/restore`).
             // See https://github.com/dotnet/msbuild/blob/a1c2e7402ef0abe36bf493e395b04dd2cb1b3540/src/MSBuild/XMake.cs#L1838
             // and https://github.com/dotnet/msbuild/issues/11519.
-            if (!noRestore)
+            if (!NoRestore)
             {
                 var restoreRequest = new BuildRequestData(
                     CreateProjectInstance(projectCollection, addGlobalProperties: static (globalProperties) =>
```

Comment:
Not new in this PR, but if there's now a restore command directly, is there some way to share this logic that's setting the environment? Otherwise it seems we risk a change to dotnet restore RealProject.csproj not being updated here?

Reply:
That's currently not possible. The real project-based restore calls msbuild.exe where this logic lives internally. Here we need to call msbuild API where the logic is not exposed. There is a tracking issue linked in the comment above to publish an API for that.